components/Modal/CreateImageUpload.js

```diff
@@ -0,0 +1,751 @@
+            break;
+          case "vsphere":
+            this.setState({ showUploadVSphereStep: true }, () => callback());
+            break;
+          default:
+            break;
+        }
+      }
+    } else if (activeStep.name === "File Upload") {
+      this.setState({ showReviewStep: true }, () => callback());
+    } else if (activeStep.name === "Review") {
+      if (this.isPendingChange()) {
+        this.handleCommit();
+      } else {
+        this.handleCreateImage();
+      }
+    } else callback();
+  }
+
+  handlePreviousStep(activeStep, callback) {
+    switch (activeStep.name) {
+      case "Review":
+        this.setState({ showReviewStep: false }, () => callback());
+        break;
+      case "Authentication":
+        this.setState(
+          {
+            showUploadAzureStep: false,
+            showUploadOpenstackStep: false,
+            showUploadVSphereStep: false
+          },
+          () => callback()
+        );
+        break;
+      default:
+        callback();
+        break;
+    }
+  }
+
+  render() {
+    const { formatMessage } = this.props.intl;
+    const { blueprint, imageTypes, providerSettings } = this.props;
+    const {
+      showUploadAzureStep,
+      showUploadOpenstackStep,
+      showUploadVSphereStep,
+      showReviewStep,
+      imageName,
+      imageType,
+      uploadService
+    } = this.state;
+
+    const imageStep = {
+      name: "Image Type",
+      component: (
+        <Form isHorizontal>
+          <FormGroup
+            label={formatMessage({ id: "blueprint-name", defaultMessage: "Blueprint" })}
+            fieldId="horizontal-form-name"
+          >
+            <Text>{blueprint.name}</Text>
+          </FormGroup>
+          <FormGroup
+            label={formatMessage({ id: "image-name", defaultMessage: "Image Name" })}
+            fieldId="horizontal-form-name"
+          >
+            <TextInput
+              value={imageName}
+              type="text"
+              id="horizontal-form-name"
+              aria-describedby="horizontal-form-name-helper"
+              name="horizontal-form-name"
+              onChange={this.setImageName}
+            />
+          </FormGroup>
+          <FormGroup
+            label={formatMessage({ id: "image-type", defaultMessage: "Type " })}
+            fieldId="horizontal-form-title"
+          >
+            <FormSelect
+              value={imageType}
+              id="horzontal-form-title"
+              name="horizontal-form-title"
+              onChange={this.setImageType}
+            >
+              <FormSelectOption isDisabled key="default" value="" label={formatMessage(messages.selectOne)} />
+              {imageTypes.map(type => (
+                <FormSelectOption isDisabled={!type.enabled} key={type.name} value={type.name} label={type.label} />
+              ))}
+            </FormSelect>
+          </FormGroup>
+          {imageType === "vhd" && (
```

Comment:
The providers code bellow has a lot of duplication. I would create a function returning the React Code for the checkboxes and call it with the different provider values.

Reply:
Fixed